### PR TITLE
[JENKINS-43990] Update to 2.X parent POM

### PR DESCRIPTION
**Pull request by varyvol**
[JENKINS-43990](https://issues.jenkins-ci.org/browse/JENKINS-43990)

- Update to 2.X parent POM.
- Bump Jenkins and plugins versions.
- Fix tests errors.

@reviewbybees @jglick @kwhetstone @andresrc @olivergondza

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -116,24 +128,6 @@
                     <argLine>-XX:MaxPermSize=128m</argLine>
```
Comment (jglick) [inline pom.xml]: Could probably be deleted, assuming we are running builds using JDK 8 going forward.

File: src/test/java/hudson/matrix/AxisTest.java [hunk 2]
```diff
@@ -140,8 +140,9 @@ private void assertFailedWith(String expected, HtmlPage res) {
     private HtmlForm addAxis(String axis) throws Exception {
         HtmlPage page = wc.getPage(p, "configure");
         HtmlForm form = page.getFormByName("config");
-        form.getButtonByCaption("Add axis").click();
+        j.getButtonByCaption(form, "Add axis").click();
         page.getAnchorByText(axis).click();
+        Thread.sleep(300);
```
Comment (jglick) [inline src/test/java/hudson/matrix/AxisTest.java]: Uh…

Comment (kwhetstone) [inline src/test/java/hudson/matrix/AxisTest.java]: I ran into this issue as well.  Glad you could just wait for it to return.

Comment (jglick) [inline src/test/java/hudson/matrix/AxisTest.java]: > As far as I know there is no waiting methods in JTH?

There are.

Comment (varyvol) [inline src/test/java/hudson/matrix/AxisTest.java]: I have implemented something simple to do the trick.

**Comment by varyvol** [comment]
@jglick @kwhetstone @olivergondza @andresrc I have changed this to implement a simple waitFor. If you could re-review, it would be great.

**Comment by varyvol** [comment]
@olivergondza I think you merged several PRs lately?

**Comment by raul-arabaolaza** [comment]
@olivergondza Any possibility of cutting a new release after this?

**Pull request merged**